app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -167,30 +139,20 @@ def manifest_history
       respond_with_template_collection(params[:action], "subscriptions", collection: @manifest_history)
     end
 
-    api :GET, "/systems/:system_id/subscriptions/available", N_("List available subscriptions"), :deprecated => true
-    param :system_id, String, :desc => N_("UUID of a content host"), :required => true
-    param :match_system, :bool, :desc => N_("Return subscriptions that match a content host")
-    param :match_installed, :bool, :desc => N_("Return subscriptions that match installed products")
-    param :no_overlap, :bool, :desc => N_("Return subscriptions which do not overlap with a currently-attached subscription")
-    def available_for_system
-      params[:match_system] = ::Foreman::Cast.to_bool(params[:match_system]) if params[:match_system]
+    def available_for_host
+      params[:match_host] = ::Foreman::Cast.to_bool(params[:match_host]) if params[:match_host]
       params[:match_installed] = ::Foreman::Cast.to_bool(params[:match_installed]) if params[:match_installed]
       params[:no_overlap] = ::Foreman::Cast.to_bool(params[:no_overlap]) if params[:no_overlap]
-      pools = @system.filtered_pools(params[:match_system], params[:match_installed],
-                                     params[:no_overlap])
-      if pools
-        available = pools.collect { |cp_pool| ::Katello::Pool.find_by(:cp_id => cp_pool['id']) }
-        available.compact!
-        available.select { |pool| pool.provider?(Organization.find(params[:organization_id])) }
-      end
-
-      available || []
+      pools = @candlepin_consumer.filtered_pools(params[:match_host], params[:match_installed], params[:no_overlap])
+      pools.select { |pool| pool.provider?(Organization.find(params[:organization_id])) }
```

Comment:
this is old code.  I'm not even sure this check is valuable.  I don't think a host would have pools from an organization it is not in.    Thoughts @thomasmckay ?